File: leetcode/0045.go
package leetcode

func jump(nums []int) int {
	n, count := len(nums), 0
	for last, mx, i := 0, 0, 0; i < n-1; i++ {
		if mx < i+nums[i] {
			mx = i + nums[i]
		}
		if last == i {
			last, count = mx, count+1
		}
	}
	return count
}

File: leetcode/0045.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size(), count = 0;
        for (int last = 0, mx = 0, cur = 0; cur < n - 1; cur++) {
            mx = max(mx, nums[cur] + cur);
            if (cur == last) {
                last = mx;
                count++;
            }
        }
        return count;
    }
};